modules/ROOT/pages/apikit-odatav4-expand-feature.adoc

```diff
@@ -0,0 +1,98 @@
+= Implement the $expand Feature
+
+The `$expand` system query option reduces the amount of incoming network traffic and makes APIs easier to consume. Because API consumers request only inline representations of the related entities of query results, the returned information avoids multiple network roundtrips. This benefits metered and latency-sensitive consumers.
+
+From APIkit for OData 1.3.0 onwards, this module supports automatic routing for ease of implementing the `$expand` system query option in the most common cases. Although this feature requires more preparation and configuration, it enables the reuse of already implemented flows when processing the `$expand` requests.
+
+For more information on the system query option, see http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#sec_SystemQueryOptionexpand[the OASIS specification^].
+
+== Requirements
+
+- APIkit for OData Module (version 1.3.0)
+- A `<request-entity-collection-listener/>` implementation for each expandable entity
+- Support for the `in` filter on the `<request-entity-collection-listener/>` flows
+
+== `$expand` Feature Example 
+
+In this example:
+
+* The `Employee` entity has a related `Manager` entity.
+* The `Employees` entity set is a collection of `Employee` entities.
+* The `Managers` entity set is a collection of `Manager` entities.
+* The API location is `https://example.com/api/`.
+
+Given this setup, the following requests retrieve employees and their managers:
+
+* Use `GET https://example.com/api/Employees` to retrieve a list of employees.
+* Gather the `Manager` IDs through the `ReferentialConstraint` of their relationship.
+* Use `GET https://example.com/api/Managers?$filter=ManagerID in (id1, id2, ..., idN)` to retrieve a list of related managers.
+* Write a small program to merge the lists based on join criteria.
+
+If using the `$expand` feature, requests are `GET https://example.com/api/Employees?$expand=Manager` to retrieve a list of employees with their `Manager` navigation property expanded.
+
+== The `generic expand` Feature
+
+The `generic expand` feature enables you to reuse existing flow implementations as a starting point for expanding requests. Using this feature, you can focus on the relevant data retrieval operations while the module performs the necessary steps to ensure that the `$expand` fields complete.
+
+An application with the `generic expand` feature enabled works as any other OData Mule application. However, during the flow execution, once the module retrieves information for the main entity set, it performs additional internal requests gathering the information for any `expand` field. If there are errors during the expansion operation, you can configure the final behavior. You can fail the request with an error, or you can replace the failed expansion with a `null` value.
```

Comment:
```suggestion
An application with the `generic expand` feature enabled works like any other OData Mule application. However, during the flow execution, once the module retrieves information for the main entity set, it performs additional internal requests to gather the information for any `expand` field. If there are errors during the expansion operation, you can configure the final behavior. You can fail the request with an error, or you can replace the failed expansion with a `null` value.
```

Reply:
Suggestion: 

Change "If there are errors during the expansion operation, you can configure the final behavior." to the following text. The current text is awkward.

You can configure how the software handles errors during the expansion operation: 
* Fail the request with an error.
* Replace the failed expansion with a `null` value.

Reply:
I like your suggestion! 
Thanks!